src/Workspaces/Core/Portable/Workspace/Solution/SolutionState.cs

```diff
@@ -1045,33 +1045,50 @@ public SolutionState WithProjectAnalyzerReferences(ProjectId projectId, IEnumera
         }
 
         /// <summary>
-        /// Create a new solution instance with the corresponding project updated to include a new 
-        /// document instanced defined by the document info.
+        /// Create a new solution instance with the corresponding projects updated to include new 
+        /// documents defined by the document info.
         /// </summary>
-        public SolutionState AddDocument(DocumentInfo documentInfo)
+        public SolutionState AddDocuments(ImmutableArray<DocumentInfo> documentInfos)
         {
-            if (documentInfo == null)
+            if (documentInfos == null)
             {
-                throw new ArgumentNullException(nameof(documentInfo));
+                throw new ArgumentNullException(nameof(documentInfos));
```

Comment:
Won't fix: the pattern here is we're actually relying on these exceptions to propagate out. We don't do the guards one layer out to avoid repeating all the checks multiple times.